Mathlib/Data/Set/ToFunction.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Data.Set.Defs
+import Mathlib.Logic.ExistsUnique
+
+/-!
+# Set to function
+
+This file provides API for sets that define functions.
+
+## Main definitions
+- `Set.asPartialFun` converts `Set (α × α)` to `α → Option α`.
+- `Set.asFun` converts `Set (α × α)` to `α → α` if possible.
+
+-/
+
+variable {α : Type*}
+
+namespace Set
+
+section set_as_partial_function
```

Comment:
No need for such a long section name
```suggestion
section partial
```

Reply:
I don't like the idea of having a section name identical to a keyword, even if it is allowed.

Reply:
As far as I know, there are no snake case sections in mathlib, and I support consistency. Also, I suggest using docs as description of the section's role. In this case, adding a 
```
/-! ### Sets as Partial Functions -/
```
before the section.